### PR TITLE
fix: disallow duplicates when updating address book

### PR DESCRIPTION
**Pull request by greatertomi**
# Checklist

- [x] JIRA - LW-7056
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Disallow duplicates when updating the address book

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/510/5320757147/index.html) for [c3b2b4ec](https://github.com/input-output-hk/lace/pull/141/commits/c3b2b4ec41139a6cb13e2df02f50511410094cc6)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 13     | 6      | 0       | 0     | 19    | ❌     |

### CODE REVIEW

**Review by gabriela-ponce — APPROVED** [review]

**Review by dmitrii-gaico — APPROVED** [review]

**Review by lucas-barros — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/features/address-book/context/AddressBookProvider.tsx [hunk 1]
```diff
@@ -32,16 +40,20 @@ export const AddressBookProvider = ({ children, initialState }: AddressBookProvi
...
-    if (hasError) {
-      toast.notify(toastParams);
-      throw new Error(toastParams.text);
-    }
+    await handleRecordValidation(list, record);
     return utils.saveRecord(record, params);
   };
 
+  const handleUpdateAddress = async (id: number, record: AddressBookSchema, params: ToastProps) => {
+    const currentList = list.filter((data) => data.id !== id);
```
Comment (VanessaPC) [inline apps/browser-extension-wallet/src/features/address-book/context/AddressBookProvider.tsx:48]: `data` needs a type.

**Review by VanessaPC — COMMENTED** [review]
Looks good to me, just a small comment.

File: apps/browser-extension-wallet/src/features/address-book/context/AddressBookProvider.tsx (hunk 1 above)
Comment (greatertomi) [reply apps/browser-extension-wallet/src/features/address-book/context/AddressBookProvider.tsx:48]: I think it gets its type from `list`.

**Review by lucas-barros — APPROVED** [review]

**Review by shawnbusuttil — APPROVED** [review]

File: apps/browser-extension-wallet/src/features/address-book/context/AddressBookProvider.tsx (hunk 1 above)
Comment (VanessaPC) [reply apps/browser-extension-wallet/src/features/address-book/context/AddressBookProvider.tsx:48]: Not for me, it's complaining about ts.

**Pull request merged**